File: Sdk-ObjectiveC/office365-base-sdk/office365-base-sdk/OAuthentication.m
//
//  OAuthentication.m
//  office365-base-sdk
//
//  Created by Gustavo on 7/17/14.
//  Copyright (c) 2014 Lagash. All rights reserved.
//

#import "OAuthentication.h"

@implementation OAuthentication : Credentials

-(id)initWith : (NSString *)token{
    self.token = token;
    return self;
};

- (void)prepareRequest:(NSMutableURLRequest *)request{
    [request addValue:[NSString stringWithFormat: @"Bearer %@", self.token] forHTTPHeaderField: @"Authorization"];
}

-(NSString *)getToken{
    return self.token;
}

@end

File: Sdk/office365_base_sdk/office365_base_sdk/OAuthentication.swift
//
//  OAuthentication.swift
//  office365-base-sdk
//
//  Created by Lagash on 6/13/14.
//  Copyright (c) 2014 microsoft. All rights reserved.
//

import Foundation

class OAuthentication : Credentials{

    var token : String;
    
    init(){
        self.token = "";
    }

    override func prepareRequest(request: NSMutableURLRequest) {
        request.addValue("Bearer " + self.token, forHTTPHeaderField: "Authorization");
    }
    
    func setToken(token :String){
        self.token = token;
    }
}